src/components/DescriptionView.jsx

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Box, Grid, IconButton, TextField, Typography } from '@mui/material';
+import Edit from '@mui/icons-material/Edit';
+import { showNotification } from 'modules/notifications';
+import UndoIcon from '@mui/icons-material/Undo';
+import SaveIcon from '@mui/icons-material/Save';
+import { withStyles } from '@mui/styles';
+import { patchDatasetDescription } from 'actions';
+import { DatasetRoles } from '../constants';
+
+const styles = (theme) => ({
+  descriptionEditor: {
+    width: '100%',
+    display: 'flex',
+  },
+  iconDiv: {
+    width: '2%',
+  },
+  textInputDiv: {
+    width: '100%',
+  },
+  descriptionInput: {
+    backgroundColor: 'white',
+  },
+  editIconButton: {
+    backgroundColor: theme.palette.primary.main,
+    border: `1px solid ${theme.palette.primary.main}`,
+    borderRadius: '3px',
+    boxShadow: 'none',
+    padding: '1px',
+    '&:hover': {
+      backgroundColor: theme.palette.primary.hover,
+      boxShadow: 'none',
+    },
+  },
+  editIconImage: {
+    color: theme.palette.primary.light,
+    fontSize: theme.typography.h6.fontSize,
+  },
+  saveDescriptionButton: {
+    boxShadow: 'none',
+    margin: theme.spacing(1),
+    '&:hover': {
+      backgroundColor: theme.palette.primary.hover,
+      boxShadow: 'none',
+    },
+  },
+  undoButton: {
+    backgroundColor: theme.palette.primary.light,
+    boxShadow: 'none',
+    color: theme.palette.grey[500],
+    margin: theme.spacing(1),
+    '&:hover': {
+      backgroundColor: theme.palette.primary.light,
+      boxShadow: 'none',
+    },
+  },
+});
+
+const maxLength = 2047;
+
+const descriptionTooLongError = {
+  response: {
+    status: 'Description too long',
+    data: {
+      message: '',
+      errorDetail: 'Exceeds 2047 characters.  Please revise and save again.',
```

Comment:
Might want to use the variable in here
```suggestion
      errorDetail: `Exceeds ${maxLength} characters.  Please revise and save again.`,
```